src/SessionDurationMetric.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+
+namespace Tightenco\NovaGoogleAnalytics;
+
+use Carbon\Carbon;
+use Illuminate\Http\Request;
+use Illuminate\Support\Arr;
+use Laravel\Nova\Metrics\Value;
+use Laravel\Nova\Metrics\ValueResult;
+use Spatie\Analytics\Analytics;
+use Spatie\Analytics\Period;
+
+class SessionDurationMetric extends Value
+{
+    public function name() {
+        return __('Avg. Session Duration');
+    }
+
+    /**
+     * Calculate the value of the metric.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return mixed
+     */
+    public function calculate(Request $request)
+    {
+        $lookups = [
+            1 => $this->sessionDurationOneDay(),
+            'MTD' => $this->sessionDurationOneMonth(),
+            'YTD' => $this->sessionDurationOneYear(),
+        ];
+
+        $data = Arr::get($lookups, $request->get('range'), ['result' => 0, 'previous' => 0]);
+
+        return (new ValueResult($data['result']))->previous($data['previous'])->format('00:00:00');
+    }
+
+    private function sessionDurationOneDay()
+    {
+        $analyticsData = app(Analytics::class)
+            ->performQuery(
+                Period::days(1),
+                'ga:avgSessionDuration',
+                [
+                    'metrics' => 'ga:avgSessionDuration',
+                    'dimensions' => 'ga:date',
+                ]
+            );
+
+        $results = collect($analyticsData->getRows());
+
+        return [
+            'result' => $results->last()[1] ?? 0,
+            'previous' => $results->first()[1] ?? 0,
+        ];
+    }
+
+    private function sessionDurationOneMonth()
+    {
+        $currentAnalyticsData = app(Analytics::class)->performQuery(
+            Period::create(Carbon::today()->startOfMonth(), Carbon::today()),
```

Comment:
Same notes/questions here, and also on the sessions metric, that I gave on the year metric for this one.